tests/functional/feature_flags/test_feature_flags.py

```diff
@@ -802,3 +802,338 @@ def test_get_configuration_with_envelope_and_raw(mocker, config):
+            },
+        }
+    }
+    feature_flags = init_feature_flags(mocker, mocked_app_config_schema, config)
+    toggle = feature_flags.evaluate(name="my_feature", context={"tenant_id": "345345435", "username": "a"}, default=False)
+    assert toggle == expected_value
+
+def test_flags_not_eqaul_match(mocker, config):
+    expected_value = True
+    mocked_app_config_schema = {
+        "my_feature": {
+            "default": expected_value,
+            "rules": {
+                "tenant id not equals 345345435": {
+                    "when_match": True,
+                    "conditions": [
+                        {
+                            "action": RuleAction.NOT_EQUALS.value,
+                            "key": "tenant_id",
+                            "value": "345345435",
+                        }
+                    ],
+                }
+            },
+        }
+    }
+    feature_flags = init_feature_flags(mocker, mocked_app_config_schema, config)
+    toggle = feature_flags.evaluate(name="my_feature", context={"tenant_id": "", "username": "a"}, default=False)
+    assert toggle == expected_value
+
+
+# Test less than
+def test_flags_less_than_no_match_1(mocker, config):
+    expected_value = False
+    mocked_app_config_schema = {
+        "my_feature": {
+            "default": expected_value,
+            "rules": {
+                "Date less than 2021.10.31": {
+                    "when_match": True,
+                    "conditions": [
+                        {
+                            "action": RuleAction.KEY_LESS_THAN_VALUE.value,
+                            "key": "current_date",
+                            "value": "2021.10.31",
+                        }
+                    ],
+                }
+            },
+        }
+    }
+    feature_flags = init_feature_flags(mocker, mocked_app_config_schema, config)
+    toggle = feature_flags.evaluate(name="my_feature", context={"tenant_id": "345345435", "username": "a", "current_date": "2021.12.25"}, default=False)
+    assert toggle == expected_value
+
+def test_flags_less_than_no_match_2(mocker, config):
+    expected_value = False
+    mocked_app_config_schema = {
+        "my_feature": {
+            "default": expected_value,
+            "rules": {
+                "Date less than 2021.10.31": {
+                    "when_match": True,
+                    "conditions": [
+                        {
+                            "action": RuleAction.KEY_LESS_THAN_VALUE.value,
+                            "key": "current_date",
+                            "value": "2021.10.31",
+                        }
+                    ],
+                }
+            },
+        }
+    }
+    feature_flags = init_feature_flags(mocker, mocked_app_config_schema, config)
+    toggle = feature_flags.evaluate(name="my_feature", context={"tenant_id": "345345435", "username": "a", "current_date": "2021.10.31"}, default=False)
+    assert toggle == expected_value
```

Comment:
perhaps merge them in a single test w/ different features?

Reply:
We could, but I always lean toward one test for one condition.

We were just bit by one test that was supposed to test several conditions and due to the complexity was not testing them but was still appearing to pass.